### Loading Uniprot IDS

In [1]:
import json

with open("proteins.json") as f:
    proteins = f.read()
    proteins = json.loads(proteins)

all_uniprot_ids = list(proteins.keys())

In [2]:
len(all_uniprot_ids)

20424

### Scraping code

In [3]:
from scrape import *

In [4]:
print(uniprot_ids[0])

A0A087X1C5


In [5]:
get_drugbank_info(uniprot_ids[0])

AttributeError: 'NoneType' object has no attribute 'find'

In [5]:
with open("protein_drugbank.json") as file:
    data = file.read()
    data = json.loads(data)

uniprot_ids = []
for k, v in data.items():
    if (v): uniprot_ids.append(k)

In [12]:
len(uniprot_ids)

3345

In [8]:
protein_drugs = {}
i = 1
for uniprot_id in uniprot_ids:
    try:
        drugs = get_drugbank_info(uniprot_id)
        protein_drugs[uniprot_id] = drugs
        print(f"{i}: Got information about {uniprot_id} in drugbank")
        i += 1
    except:
        protein_drugs[uniprot_id] = None
        print(f"No information about {uniprot_id} in drugbank")

# 408-409 issue!

https://go.drugbank.com/bio_entities/BE0009860
1: Got information about A5X5Y0 in drugbank
https://go.drugbank.com/bio_entities/BE0003619
2: Got information about A8MPY1 in drugbank
https://go.drugbank.com/bio_entities/BE0009802
3: Got information about C9JRZ8 in drugbank
https://go.drugbank.com/bio_entities/BE0004105
4: Got information about O00141 in drugbank
https://go.drugbank.com/bio_entities/BE0004639
5: Got information about O00142 in drugbank
https://go.drugbank.com/bio_entities/BE0000154
6: Got information about O00180 in drugbank
https://go.drugbank.com/bio_entities/BE0004333
7: Got information about O00182 in drugbank
https://go.drugbank.com/bio_entities/BE0004630
8: Got information about O00187 in drugbank
https://go.drugbank.com/bio_entities/BE0001264
9: Got information about O00204 in drugbank
https://go.drugbank.com/bio_entities/BE0002442
10: Got information about O00206 in drugbank
https://go.drugbank.com/bio_entities/BE0000037
11: Got information about O00217 in drugba

In [9]:
json_data = json.dumps(protein_drugs)
with open("protein_drugbank.json", "w") as file:
    file.write(json_data)
file.close()

### Redoing the second time

In [2]:
import json

with open("protein_drugbank.json") as file:
    d = file.read()
    d = json.loads(d)

In [6]:
len(d.items())

3345

In [24]:
for k,v in d.items():
    if not v:
        print(k)
        d[k] = get_drugbank_info(k)
        print(d[k])

P23946
https://go.drugbank.com/bio_entities/BE0001038
{'DB03297': ['Benzylsulfonic acid', 'experimental', 'unknown'], 'DB03814': ['2-(N-morpholino)ethanesulfonic acid', 'experimental', 'unknown'], 'DB04016': ['2-[3-({Methyl[1-(2-Naphthoyl)Piperidin-4-Yl]Amino}Carbonyl)-2-Naphthyl]-1-(1-Naphthyl)-2-Oxoethylphosphonic Acid', 'experimental', 'unknown'], 'DB07680': ['[(1S)-1-(5-CHLORO-1-BENZOTHIEN-3-YL)-2-(2-NAPHTHYLAMINO)-2-OXOETHYL]PHOSPHONIC ACID', 'experimental', 'unknown']}
P46781
https://go.drugbank.com/bio_entities/BE0009117
{'DB11638': ['Artenimol', 'approved, experimental, investigational', 'unknown', 'ligand']}
P50213
https://go.drugbank.com/bio_entities/BE0000011
{'DB00157': ['NADH', 'approved, nutraceutical', 'unknown'], 'DB09130': ['Copper', 'approved, investigational', 'unknown'], 'DB06757': ['Manganese', 'approved, nutraceutical', 'unknown'], 'DB09092': ['Xanthinol', 'approved, withdrawn', 'yes', 'cofactor']}
Q8N5S9
https://go.drugbank.com/bio_entities/BE0009418
{'DB12010': 

In [25]:
json_data = json.dumps(d)
with open("protein_drugbank.json", "w") as file:
    file.write(json_data)
file.close()

#### It may be possible that previously left out Total - 3345 also had some of these type of issues

In [7]:
with open("protein_drugbank.json") as file:
    d = file.read()
    d = json.loads(d)

In [8]:
druggable_uniprot_ids = list(d.keys())
len(druggable_uniprot_ids)

3345

In [8]:
for uniprot_id in all_uniprot_ids[:2000]:
    if uniprot_id not in druggable_uniprot_ids:
        try:
            get_drugbank_info(uniprot_id)
            print(uniprot_id)
        except:
            print("Yes")

Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
